napi.h

```diff
@@ -1515,7 +1515,20 @@ namespace Napi {
 
   protected:
     explicit AsyncWorker(const Function& callback);
-    explicit AsyncWorker(const Object& receiver, const Function& callback);
+    AsyncWorker(const Function& callback,
+                const char* resource_name);
+    AsyncWorker(const Function& callback,
+                const char* resource_name,
+                const Object& resource);
+    AsyncWorker(const Object& receiver,
+                const Function& callback);
+    AsyncWorker(const Object& receiver,
+                const Function& callback,
+                const char* resource_name);
+    AsyncWorker(const Object& receiver,
+                const Function& callback,
+                const char* resource_name,
+                const Object& resource);
```

Comment:
Overall looks good but do we need to add explicit (like there was for the existing methods) to avoid unwanted conversions?